packages/app-builder/src/services/validation/ast-node-validation.ts

```diff
@@ -146,30 +146,44 @@ export function findArgumentErrorsFromParent(
   }
 }
 
-export function getValidationStatus({
-  evaluationErrors,
-  valueIsNull,
-  isDivByZeroField,
+export function getIndirectEvaluationErrors({
   parentEvaluationErrors,
   pathSegment,
+  isDivByZeroField,
+  valueIsNull,
 }: {
-  evaluationErrors: EvaluationError[];
-  valueIsNull: boolean;
-  isDivByZeroField: boolean;
   parentEvaluationErrors?: EvaluationError[];
   pathSegment?: PathSegment;
-}): ValidationStatus {
-  if (evaluationErrors.length > 0) return 'error';
-
+  isDivByZeroField: boolean;
+  valueIsNull: boolean;
+}): EvaluationError[] {
+  const out = [] as EvaluationError[];
   if (pathSegment && parentEvaluationErrors) {
-    const argumentErrors = findArgumentErrorsFromParent(
-      pathSegment,
-      parentEvaluationErrors,
+    out.push(
+      ...findArgumentErrorsFromParent(pathSegment, parentEvaluationErrors),
     );
-    if (argumentErrors.length > 0) return 'light-error';
   }
-  if (isDivByZeroField) return 'light-error';
-  if (valueIsNull) return 'light-error';
+  if (isDivByZeroField) {
+    out.push({
+      error: 'DIVISION_BY_ZERO',
+      message: 'Division by zero error',
+    });
+  }
+  if (valueIsNull) {
+    out.push({ error: 'NULL_FIELD_READ', message: 'Null value read' });
+  }
+  return out;
+}
+
+export function getValidationStatus({
+  evaluationErrors,
+  indirectEvaluationErrors,
+}: {
+  evaluationErrors: EvaluationError[];
+  indirectEvaluationErrors: EvaluationError[];
+}): ValidationStatus {
+  if (evaluationErrors.length > 0) return 'error';
+  if (indirectEvaluationErrors.length > 0) return 'light-error';
```

Comment:
That's not exactly what I have in mind.

The idea is to capitalize on the existing error structure :
1. direct error
2. argument error (from parent)
3. valid

With the `"FUNCTION_ERROR"` example, we modify the returned error tree from the backend validation to add specifc errors to the frontend (based on the error structure)

Here, the idea would be the same, build upfront the error tree using this logic. 
Based on the current implementation, it may be an argument error added to the parent.
Probably something like this :
1. I have a `division_by_zero` error
2. Traverse the tree to find possible error nodes
3. add an argument error when the error condition is true

Reply:
👍  as discussed, we merge as it is and review the underlying design tomorrow